src/html5/parser.rs

```diff
@@ -1033,15 +1055,18 @@ impl<'chars> Html5Parser<'chars> {
 
                         if process_as_intable_anything_else {
                             let tmp = self.current_token.clone();
-                            self.current_token = Token::Text(tokens);
-
                             self.foster_parenting = true;
-                            self.handle_in_body();
-                            self.foster_parenting = false;
 
+                            let tokens = self.split_mixed_token(&pending_chars);
```

Comment:
special case where we split tokens manually. After this, we call handle_in_body for each of those tokens before continuing the rest of the flow.